front/components/assistant/conversation/ConversationMessage.tsx

```diff
@@ -1,35 +1,53 @@
-import { Avatar } from "@dust-tt/sparkle";
+import { Avatar, IconButton } from "@dust-tt/sparkle";
+import { ComponentType, MouseEventHandler } from "react";
 
 /**
  * Parent component for both UserMessage and AgentMessage, to ensure avatar,
  * side buttons and spacing are consistent between the two
  */
 export function ConversationMessage({
+  messageId,
   children,
   name,
   pictureUrl,
   buttons,
 }: {
   children?: React.ReactNode;
   name: string | null;
+  messageId: string;
   pictureUrl?: string | null;
-  buttons?: React.ReactNode[];
+  buttons?: {
+    icon: ComponentType;
+    onClick: MouseEventHandler<HTMLButtonElement>;
+  }[];
 }) {
   return (
-    <div className="flex w-full flex-row gap-4">
-      <div className="flex-shrink-0">
-        <Avatar visual={pictureUrl} name={name || undefined} size="sm" />
-      </div>
-      <div className="min-w-0 flex-grow">
-        <div className="flex flex-col gap-4">
-          <div className="text-sm font-medium">{name}</div>
-          <div className="min-w-0 break-words text-base font-normal">
-            {children}
+    <div className="mx-auto flex max-w-4xl gap-4">
```

Comment:
Doesn't it break the background that supposed to take the full width?

Reply:
The idea is that User/Agent message can render anywhere and should adapt to the layout they are asked in